File: algorithms_and_data_structures/lecture-2/bubble_sort.py
# Сортировка пузырьком
# Сложность алгоритма = O(n^2)


def bubble_sort(data):
    for i in range(len(data)-1):
        for j in range(len(data)-i-1):
            if data[j] > data[j+1]:
                data[j], data[j+1] = data[j+1], data[j]
    return data


array = [3, 5, 1, 8, 2, 9, 0, 4, 6]
print(bubble_sort(array))

File: algorithms_and_data_structures/lecture-2/bubble_sort.java
public class bubble_sort {
    public static void main(String[] args) {
        int[] array = new int[] {
                4, 2, 5, 8, 1, 9, 2, 3, 6, 8, 5
        };

        bubble_sorting(array);

        for (int i = 0; i < array.length; i++) {
            System.out.print(array[i] + " ");
        }
    }

    public static void bubble_sorting(int[] array) {
        boolean finish;
        do {
            finish = true;
            for (int i = 0; i < array.length - 1; i++) {
                if (array[i] > array[i + 1]) {
                    int temp = array[i];
                    array[i] = array[i + 1];
                    array[i + 1] = temp;
                    finish = false;
                }
            }
        } while (!finish);

    }
}